pkg/runtime/wfengine/workflow.go

```diff
@@ -175,12 +176,17 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 		return errors.New("invalid execution start event")
 	}
 
-	// We block (re)creation of existing workflows unless they are in a completed state.
+	// We block (re)creation of existing workflows unless they are in a completed state
+	// Or if they still have any pending activity result awaited.
 	if !created {
```

Comment:
change flow as such:
```go
  if created {
     return error
  }
  // unnest the rest 
  runtimeState := getRuntimeState(actorID, state)
  if !runtimeState.IsCompleted() {
     return error
  }
  if wf.activityResultAwaited {
     return fmt.Errorf("a terminated workflow with ID '%s' is already awaiting an activity result", actorID)
  }
  // no else needed
  wfLogger.Infof("%s: workflow was previously completed and is being recreated", actorID)
  state.Reset()
```

Reply:
Could not take the first check from suggestion:
```
  if created {
     return error
  }
```
I have taken the later part of suggestion, that sounds good.